diff -urN a/xen/arch/x86/Rules.mk b/xen/arch/x86/Rules.mk
--- a/xen/arch/x86/Rules.mk     2021-04-22 01:58:30.249432277 +0530
+++ b/xen/arch/x86/Rules.mk     2021-04-22 02:01:52.995150246 +0530
@@ -7,6 +7,7 @@
 CFLAGS += -I$(BASEDIR)/include/asm-x86/mach-generic
 CFLAGS += -I$(BASEDIR)/include/asm-x86/mach-default
 CFLAGS += -DXEN_IMG_OFFSET=$(XEN_IMG_OFFSET)
+CFLAGS += '-U__OBJECT_LABEL__'
 CFLAGS += '-D__OBJECT_LABEL__=$(subst /,$$,$(subst -,_,$(subst $(BASEDIR)/,,$(CURDIR))/$@))'

 # Prevent floating-point variables from creeping into Xen.
diff -urN a/xen/Rules.mk b/xen/Rules.mk
--- a/xen/Rules.mk      2021-04-22 01:58:30.273431564 +0530
+++ b/xen/Rules.mk      2021-04-22 01:56:33.723648591 +0530
@@ -61,6 +61,7 @@
 $(call cc-option-add,CFLAGS,CC,-Wvla)
 CFLAGS += -pipe -D__XEN__ -include $(BASEDIR)/include/xen/config.h
 CFLAGS-$(CONFIG_DEBUG_INFO) += -g
+CFLAGS += '-U__OBJECT_FILE__'
 CFLAGS += '-D__OBJECT_FILE__="$@"'

 ifneq ($(clang),y)
 